dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -746,21 +748,25 @@ HRESULT WINAPI CRecycleBin::GetDetailsOf(PCUITEMID_CHILD pidl, UINT iColumn, LPS
             FormatDateTime(buffer, MAX_PATH, &pFileDetails->LastModification);
             break;
         case COLUMN_TYPE:
-            // FIXME: We should in fact use a UNICODE version of _ILGetFileType
-            szTypeName[0] = L'\0';
-            wcscpy(buffer, PathFindExtensionW(pFileDetails->szName));
-            if (!( HCR_MapTypeToValueW(buffer, buffer, _countof(buffer), TRUE) &&
-                    HCR_MapTypeToValueW(buffer, szTypeName, _countof(szTypeName), FALSE )))
+            SEARCH_CONTEXT Context;
+            Context.pFileDetails = pFileDetails;
+            Context.bFound = FALSE;
+            EnumerateRecycleBinW(NULL, CBSearchRecycleBin, (PVOID)&Context);
+            
```

Comment:
Please configure your code editor to systematically trim any trailing whitespace (and make it show the whitespace).